libs/akita/src/lib/transaction.ts

```diff
@@ -86,16 +87,16 @@ export function applyTransaction<T>(action: () => T, thisArg = undefined): T {
  *
  */
 export function transaction() {
-  return function(target: any, propertyKey: string, descriptor: PropertyDescriptor) {
-    const originalMethod = descriptor.value;
+  return (target: any, propertyKey: string, descriptor: PropertyDescriptor): PropertyDescriptor => {
+    const descriptorCopy = { ...descriptor };
```

Comment:
We don't really need a copy

Reply:
I made a copy because of `descriptor.value = ...` that's mutating the input parameter
https://eslint.org/docs/rules/no-param-reassign

There are other ways to fix it. But mutation should be avoided